web/src/api/storage/types/checks.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import * as config from "./config";
+
+// Type guards.
+
+export function isFormattedDrive(drive: config.DriveElement): drive is config.FormattedDrive {
+  return "filesystem" in drive;
+}
+
+export function isPartitionedDrive(drive: config.DriveElement): drive is config.PartitionedDrive {
+  return !("filesystem" in drive);
+}
+
+export function isSearchAll(search: config.Search): search is config.SearchAll {
```

Comment:
The name of this function can be confusing. There are many ways to search for all.

But this functions checks whether a particular way (the asterisk which is literally a shortcut for `{"search": { "ifNotFound": "skip" }}`) is used.

I would prefer a name that reflects that. I don't have a good suggestion, though. Maybe `isSearchAllShortcut` or something along that line.

Reply:
I know it corresponds to the name of the type `searchByName` but I still find it kind of confusing. Maybe a better name for the type, maybe a better name for this function... I don't know.

Reply:
Yes, it is tricky to find good names for that types. If we change it, I would change the type name. IMHO, changing the type guard name is not a good idea.